subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -39,8 +37,7 @@ class TaskInputPropertiesIntegrationTest extends AbstractIntegrationSpec {
         """
 
         when: fails "foo"
-        then: failure.assertHasDescription("Could not add entry ':foo' to cache taskHistory.bin")
-        then: failure.assertHasCause("Unable to store task input properties. Property 'b' with value 'xxx' cannot be serialized.")
+        then: failure.assertHasDescription("Unable to hash task input properties. Property 'b' with value 'xxx' cannot be serialized.")
```

Comment:
I have the feeling that removing the call to `TaskHistoryStore#flush()` isn't as simple as it may be. Given the following comment in the code:
```
// This is not strictly necessary, as the caches are closed immediately after this. Calling flush here rethrows any write failures inside the context of the build
// Instead, failures thrown when stopping or closing a service should be treated as build failures
gradle.getServices().get(TaskHistoryStore.class).flush();
```

It seems to imply the `flush()` is required to collect any write failures and treat them as build failures. Removing that call may result in some feature loose. I would suggest addressing this independently of this PR as more code would need to be changed somewhat unrelated to this code.